### PR TITLE
Update Dependencies to Latest

### PR DESCRIPTION
**Pull request by JT122406**
Yes I verified this builds on my system and shouldn't have any breaking changes mainly just security changes

### CODE REVIEW

**Comment by JT122406** [comment]
This one builds on System and should be fine in WorkFlow

File: build.gradle [hunk 1]
```diff
@@ -10,13 +10,13 @@ allprojects {
 
     ext {
         // dependency versions
-        adventureVersion = '4.10.1'
-        junitVersion = '5.7.0'
-        slf4jVersion = '1.7.30'
-        log4jVersion = '2.17.1'
+        adventureVersion = '4.11.0'
+        junitVersion = '5.9.0'
+        slf4jVersion = '2.0.1'
```
Comment (Kichura) [inline build.gradle]: SLF4J 2.0.2 is available by the way.

Comment (JT122406) [reply build.gradle]: ah ok thanks

**Comment by astei** [comment]
A good chunk of these are breaking. I took only the ones not likely to break the public API and put them in 7d77bfb5

**Comment by A248** [comment]
By the way, Slf4j 2.0 is binary compatible with respect to client code using Slf4j 1.7. Only the logging framework would need to be updated -- that is, `log4j-slf4j-impl` needs to support Slf4j 2.0. The Slf4j docs describe the changes in 2.0.x: https://www.slf4j.org/faq.html#changesInVersion200.

Something to consider perhaps for another time.

**Pull request closed without merging**